packages/rest/src/parser.ts

```diff
@@ -78,29 +93,58 @@ export async function parseOperationArgs(
 async function loadRequestBodyIfNeeded(
   operationSpec: OperationObject,
   request: Request,
-): Promise<MaybeBody> {
-  if (!operationSpec.requestBody) return Promise.resolve();
+  options: RequestBodyParserOptions = {},
+): Promise<RequestBody> {
+  if (!operationSpec.requestBody) return Promise.resolve({value: undefined});
+
+  debug('Request body parser options: %j', options);
 
   const contentType = getContentType(request);
   debug('Loading request body with content type %j', contentType);
+
+  if (
+    contentType &&
+    contentType.startsWith('application/x-www-form-urlencoded')
+  ) {
+    const body = await parseFormBody(request, options).catch(
```

Comment:
I find it rather weird that you are mixin `await` and `.catch()` style. Can you rewrite this using try/catch please?

```ts
try {
  const body = await parseFormBody(request, options)
  return { /* ... */ };
} catch (err) {
  // handle the error, use a helper function shared by JSON & form parsers
  throw normalizeBodyParserError(err);
}
```